### DISCUSSION TITLE
Enhancement: Support "name" claim in OpenID

### DISCUSSION THREAD

**Discussion by alihacks** [danny-avila/LibreChat#4244] [Feature Requests & Suggestions]
### What features would you like to see added?

Support "name" claim in OpenID if given_name and family_name are not present

### More details

Our OpenID provider does not provide separate first and last names, but provides a "name" claim which is the user's full name. I would like to submit a PR that changes the fallback logic to use the "name" claim if first and last name are not available before falling back to using the username or email address for full name.

### Which components are impacted by your request?

General

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
PRs welcome. Writing tests for existing behavior would be appreciate before any further edits to openidstrategy as a lot of people rely on it across different providers
